utils/_weblog.py

```diff
@@ -78,7 +79,10 @@ def request(
                 break
 
         user_agent = headers.get(user_agent_key, "system_tests")
-        headers[user_agent_key] = f"{user_agent} rid/{rid}"
+        # Inject a request id (rid) to be able to correlate traces generated from this request.
+        # This behavior can be disabled with rid_in_user_agent=False, which should be rarely needed.
+        if rid_in_user_agent:
```

Comment:
This hack wouldn't be needed once all libraries are updated with new rules: https://github.com/DataDog/appsec-event-rules/pull/150
but it will be quite some time until that gets in.